autosklearn/data/validation.py

```diff
@@ -239,8 +263,25 @@ def _validate_dataframe_features(
         Uses .iloc as a safe way to deal with pandas object
         """
 
+        # No Nan is supported
+        if np.any(pd.isnull(X)):
+            raise ValueError("Categorical features array cannot contain missing/NaN values. "
+                             "You can pre-process your data via: "
+                             "https://scikit-learn.org/stable/modules/impute.html "
+                             "before feeding it to auto-sklearn."
+                             )
+
         # Start with the features
-        enc_columns, feature_types = self._check_and_get_columns_to_encode(X)
+        if hasattr(X, "iloc"):
+            enc_columns, feature_types = self._check_and_get_columns_to_encode(X)
+        else:
+            if len(X.shape) < 1:
+                raise ValueError("Expected features to have more than 1 dimensionality"
```

Comment:
Could you please make this more concrete, i.e. it needs to be exactly 2d for Auto-sklearn.

Reply:
Done in next push!